tac-operation-lifecycle/tac-operation-lifecycle-logic/src/database.rs

```diff
@@ -0,0 +1,1218 @@
+            }
+
+            Err(e) => {
+                tx.rollback().await?;
+                tracing::error!("Failed to add pending interval [{}..{}]: {}", from, to, e);
+                return Err(e.into());
+            }
+        };
+
+        Ok(())
+    }
+
+    // Create intervals with the specified period within the [from..to] boundary
+    // Returns the number of created intervals
+    // NOTE: the routine doesn't check if intervals will overlapped with existing ones
+    pub async fn generate_pending_intervals(
+        &self,
+        from: u64,
+        to: u64,
+        period_secs: u64,
+    ) -> anyhow::Result<usize> {
+        const DB_BATCH_SIZE: usize = 1000;
+        let now = chrono::Utc::now().naive_utc();
+        
+        let intervals: Vec<interval::ActiveModel> = (from..to)
+            .step_by(period_secs as usize)
+            .map(|timestamp| {
+                let start_naive = Self::timestamp_to_naive(timestamp as i64);
+                let finish_naive = Self::timestamp_to_naive(min(timestamp + period_secs, to) as i64);
+                
+                interval::ActiveModel {
+                    id: ActiveValue::NotSet,
+                    start: ActiveValue::Set(start_naive),
+                    //we don't want to save intervals that are out of specifiedboundaries
+                    finish: ActiveValue::Set(finish_naive),
+                    inserted_at: ActiveValue::Set(now),
+                    updated_at: ActiveValue::Set(now),
+                    status: sea_orm::ActiveValue::Set(EntityStatus::Pending.to_id()),
+                    next_retry: ActiveValue::Set(None),
+                    retry_count: ActiveValue::Set(0 as i16),
+                }
+            })
+            .collect();
+
+        tracing::debug!(
+            "Total {}s intervals were generated: {} ({}..{})",
+            period_secs,
+            intervals.len(),
+            from,
+            to
+        );
+
+        // Process intervals in batches
+        for chunk in intervals.chunks(DB_BATCH_SIZE) {
+            let tx = self.db.begin().await?;
+
+            match interval::Entity::insert_many(chunk.to_vec())
+                .exec_with_returning(&tx)
+                .await
+            {
+                Ok(_) => {
+                    // Update watermark to the end of the last interval in this batch
+                    // [assume bathes are sorted by `start` field ascending at that point]
+                    let mut updated_watermark: Option<u64> = None;
+                    if let Some(last_interval_from_batch) = chunk.last() {
+                        if let ActiveValue::Set(finish) = &last_interval_from_batch.finish {
+                            updated_watermark = Some(finish.and_utc().timestamp() as u64);
+                            let _ = self
+                                .set_watermark_internal(&tx, updated_watermark.unwrap())
+                                .await;
```

Comment:
you dont need unwrap here
```suggestion
                            timestamp = finish.and_utc().timestamp() as u64;
                            let _ = self
                                .set_watermark_internal(&tx, updated_watermark)
                                .await;
                            updated_watermark = Some(timestamp);
```

Reply:
Fixed in https://github.com/blockscout/blockscout-rs/pull/1325/commits/03b4b3186f5e0327e72f729bef048bc117a69546